### PR TITLE
Update docs and add new use-cases as quickstart guides

### PR DESCRIPTION
**Pull request by AlexIoannides**
The docs have been updated as follows:

* the `Overview` page has been re-worked to make it much easier to understand what Bodywork does in the first couple of sentences.
* The `Key Concepts` page has had `Example: Batch Job` and `Example: Serve Model` sections added.
* A new `Quickstart - Batch Job` page has been added.
* A new `Quickstart - Serve Model` page has been added.
* A new `Template Projects` page has been added.
* All pages have been reviewed and where necessary edited to make sentences shorter and paragraphs easier to read.

[overview.pdf](https://github.com/bodywork-ml/bodywork-core/files/5758946/overview.pdf)
[installation.pdf](https://github.com/bodywork-ml/bodywork-core/files/5758944/installation.pdf)
[key_concepts.pdf](https://github.com/bodywork-ml/bodywork-core/files/5758945/key_concepts.pdf)
[quickstart_batch_job.pdf](https://github.com/bodywork-ml/bodywork-core/files/5758947/quickstart_batch_job.pdf)
[quickstart_serve_model.pdf](https://github.com/bodywork-ml/bodywork-core/files/5758948/quickstart_serve_model.pdf)
[quickstart_train_deploy.pdf](https://github.com/bodywork-ml/bodywork-core/files/5758949/quickstart_train_deploy.pdf)
[user_guide.pdf](https://github.com/bodywork-ml/bodywork-core/files/5758951/user_guide.pdf)
[cli_reference.pdf](https://github.com/bodywork-ml/bodywork-core/files/5758943/cli_reference.pdf)
[template_projects.pdf](https://github.com/bodywork-ml/bodywork-core/files/5758950/template_projects.pdf)

### CODE REVIEW

File: docs/cli_reference.md [hunk 1]
```diff
@@ -96,16 +96,16 @@ Will list information on all active service deployments available in `YOUR_NAMES
...
+    --namespace=YOUR_NAMESPACE \
     --name=SERVICE_NAME
 ```
 
 Delete an active service deployment - e.g. one that is no longer required for a project.
 
 ## Manage Cronjobs
 
-Bodywork can schedule workflows to run periodically to a pre-defined schedule, using remote workflow-controllers.
+Bodywork can schedule workflows to run periodically on a schedule. These workflows are managed by workflow-controllers started on your k8s cluster.
```
Comment (Marios85) [inline docs/cli_reference.md:108]: Maybe make it clearer that these workflow controllers are automatically deployed/started by bodywork.

e.g. These workflows are managed by the workflow-controllers automatically started on your k8s cluster by bodywork.

File: docs/quickstart_batch_job.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+EXECUTABLE_SCRIPT="score.py"
+CPU_REQUEST=0.5
+MEMORY_REQUEST_MB=100
+
+[batch]
+MAX_COMPLETION_TIME_SECONDS=30
+RETRIES=2
+```
+
+From which it is clear to see that we have specified that this stage is a batch stage (as opposed to a service-deployment), that `score.py` should be the script that is run, together with an estimate of the CPU and memory resources to request from the k8s cluster, how long to wait and how many times to retry, etc.
```
Comment (Marios85) [inline docs/quickstart_batch_job.md:105]: If I remember correctly, defaults are used if these are not specified. If so then specify that defaults will be used if they're not present.

File: docs/quickstart_serve_model.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+## What am I going to Learn?
+
+* How to configure a Bodywork project to deploy a pre-trained model, as a model-scoring service with a REST API, on Kubernetes.
+* How to deploy the service.
+* How to test the deployed service.
+
+## A ML Scoring Service - Returning the Predicted Class
+
+The example ML model that we want to expose as a service with a REST API, returns the predicted sub-species of an iris plant, given four of its critical dimension as inputs. For more information on this ML task see ['Quickstart - Train and Deploy Pipeline'](quickstart_train_and_deploy.md#a-machine-learning-task).
```
Comment (Marios85) [inline docs/quickstart_serve_model.md:15]: dimension**s**

**Review by Marios85 — COMMENTED** [review]
Just a couple of in-line comments. It's clear and easy to follow, adding the templates was a really good idea.

File: docs/cli_reference.md (hunk 1 above)
Comment (AlexIoannides) [reply docs/cli_reference.md:108]: Agreed.

File: docs/quickstart_batch_job.md (hunk 2 above)
Comment (AlexIoannides) [reply docs/quickstart_batch_job.md:105]: Just checked the source code - no defaults are used.

File: docs/quickstart_serve_model.md (hunk 3 above)
Comment (AlexIoannides) [reply docs/quickstart_serve_model.md:15]: Well spotted!

**Review by Marios85 — APPROVED** [review]
😎

**Pull request merged**